CDCgov/prime-reportstream
# 7933 Submission history API supports bulk FHIR submissions

This PR updates the Submission History API to support bulk FHIR submissions. Currently the API would return each report as a separate destination. SubmissionHistory was updated to consolidate all the reports into their respective destination.

Test Steps:
1. Send a bulk FHIR submission with two or more bundles via the reports API.
2. Call the Submission History API with the report id you submitted
3. Verify bundles are being routed to the right destination, and no duplicate destinations are created

## Changes
- Updated `enrichWithRoutingAndTranslationActions` in `SubmissionHistory.kt` to handle multiple reports
- Added unit tests for new added functionality

## Checklist

### Testing
- [ x Added tests?

## Linked Issues
- Fixes #7933
